src/main/java/database/service/service/JsonDatabaseService.java

```diff
@@ -0,0 +1,195 @@
+        entities.add(entity);
+
+        objectMapper.writeValue(databasePath.toFile(), entities);
+        return entity;
+    }
+
+    @Override
+    public <T extends BaseEntity, I> T updateRecordInTable(T entity, I id) throws IOException,
+            ProvidedIdDoesNotExistException {
+        Class<? extends BaseEntity> entityClass = entity.getClass();
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        List<T> entities = deserializeEntities(entityClass, Files.readString(databasePath));
+
+        Optional<T> optionalEntity = entities.stream().filter(e -> id.equals(e.getId())).findFirst();
+        T entityFoundById;
+        if (optionalEntity.isPresent()) {
+            entityFoundById = optionalEntity.get();
+            updateEntityFields(entityFoundById, entity);
+        } else {
+            throw new ProvidedIdDoesNotExistException("Entity with provided Id does not exist.");
+        }
+
+        objectMapper.writeValue(databasePath.toFile(), entities);
+        return entityFoundById;
+    }
+
+    @Override
+    public <I> boolean removeRecordFromTable(Class<? extends BaseEntity> entityClass, I id) throws IOException {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        List<? extends BaseEntity> entities = deserializeEntities(entityClass, Files.readString(databasePath));
+
+        boolean isEntityRemoved = entities.removeIf(e -> id.equals(e.getId()));
+
+        objectMapper.writeValue(databasePath.toFile(), entities);
+        return isEntityRemoved;
+    }
+
+    @Override
+    public void removeAllRecordsFromTable(Class<? extends BaseEntity> entityClass) throws IOException {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+        Files.writeString(databasePath, "[]");
+    }
+
+    @Override
+    public <T extends BaseEntity, I> T getById(Class<? extends BaseEntity> entityClass, I id) throws IOException,
+            ProvidedIdDoesNotExistException {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        List<T> entities = deserializeEntities(entityClass, Files.readString(databasePath));
+
+        return entities.stream()
+                .filter(e -> id.equals(e.getId()))
+                .findFirst()
+                .orElseThrow(() -> new ProvidedIdDoesNotExistException("Entity with provided Id does not exist."));
+    }
+
+    @Override
+    public <T extends BaseEntity> Iterable<T> getAllRecordsFromTable(Class<? extends BaseEntity> entityClass)
+            throws IOException {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        return deserializeEntities(entityClass, Files.readString(databasePath));
+    }
+
+    @Override
+    public <T extends BaseEntity, V> Iterable<T> getByFilters(Class<? extends BaseEntity> entityClass,
+                                                              Set<Pair<String, V>> filters) throws IOException {
```

Comment:
Please add validation for `filter`. It can store incorrect, empty or null keys and values. Consider creating `Validator` class in `helper` package.